### DISCUSSION TITLE
Offchain cancel offer issue

### DISCUSSION THREAD

**Discussion by farairo** [ProjectOpenSea/opensea-js#1540] [General]
I am trying to implement offchain cancel offer feature.  It works, when the offerer and the apiKey match, but it does not when I sign a message from the offerer and include that signature to a request with an apiKey from a different wallet. It complains with apiKey is not from offerer, although signature is valid. So my question is does cancel request works with just offererSignature or we need offererSignature  and api key from same wallet to cance the offer? 
here is my code
`const domain = {
        name: SEAPORT_CONTRACT_NAME,
        version: SEAPORT_CONTRACT_VERSION_V1_6,
        chainId: "1",
        verifyingContract: CROSS_CHAIN_SEAPORT_V1_6_ADDRESS,
      };

      // Define the types structure
      const types = {
        OrderHash: [{ name: "orderHash", type: "bytes32" }],
      };

      // Define the value to sign (order hash)
      const value = {
        orderHash: orderHash, // The bytes32 order hash
      };
      let sign = await signer.signTypedData(domain, types, value);
      console.log("sign", sign);
      let body = {
        offererSignature: sign,
      };

      let response1 = await axios.post(
        `https://api.opensea.io/api/v2/orders/chain/ethereum/protocol/0x0000000000000068f116a894984e2db1123eb395/${orderHash}/cancel`,
        body,
        {
          //proxy: false,
          //httpsAgent: useProxy ? proxy.proxy : null,
          headers: {
            "Content-Type": "application/json",
            "X-API-KEY": openSeaApiKey,
          },
        }
      );`
The docs here https://docs.opensea.io/reference/cancel_order says it should work with either one but does not seems to work for me.
Any help will be really appreciated

**Comment by ryanio** [comment]
the offererSignature will work without the corresponding api key.
have you tried the `offchainCancelOrder` method in opensea-js? your code seems similar but wondering if there is something i'm not seeing
i would also double check your signer address is the same as the order "offerer" field

**Reply by farairo** [reply]
Thanks for the reply.
could you please link to offchainCancelOrder docs.
Regarding signer address yes its same as order offerer address but still getting "API key not from order's maker".
maybe my offererSignature is not right?
here is part of my code where i set offerer address 

`let orderComponents = {

    offerer: offerer, // this address is same as signer address while canceling offer

    zone: "0x000056f7000000ece9003ca63978907a00ffd100",

    offer: offer,

    consideration: consideration,

    orderType: 3,

    startTime: Math.round(Date.now() / 1000).toString(),

    endTime: exp_time.toString(),

    zoneHash:
      "0x0000000000000000000000000000000000000000000000000000000000000000",

    salt: generateRandomSalt(),

    conduitKey:
      "0x0000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000",

    counter: 0,

    //restrictedByZone: false,

    //allowPartialFills: true,

  };`

**Reply by ryanio** [reply]
There is an offchainCancelOrder method, no docs but here's link to source, try this: https://github.com/ProjectOpenSea/opensea-js/blob/main/src/sdk.ts#L980

**Reply by farairo** [reply]
ok thanks for the link, i will check it out.
is there any way to check if signature is correct?
